File: php/testcode/headerinj_006.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_headerinj_cookie_safe
function headerinj006(BenchmarkRequest $req): BenchmarkResponse {
    $allowedNames = ['theme', 'lang', 'timezone'];
    $name = $req->param('name');
    $value = $req->param('value');
    if (in_array($name, $allowedNames, true)) { // vuln-code-snippet safe-line php_headerinj_cookie_safe
        setcookie($name, urlencode($value));
        return BenchmarkResponse::ok('Cookie set');
    }
    return BenchmarkResponse::badRequest('Invalid cookie name');
}
// vuln-code-snippet end php_headerinj_cookie_safe

File: ruby/testcode/headerinj_006.rb
require_relative 'shared'

require 'uri'

# vuln-code-snippet start ruby_headerinj_encode_value
def set_custom_header(req)
  raw_value = req.param('data')
  encoded = URI.encode_www_form_component(raw_value)
  headers = { 'X-Custom-Data' => encoded }  # vuln-code-snippet safe-line ruby_headerinj_encode_value
  BenchmarkResponse.new(200, 'ok', headers)
end
# vuln-code-snippet end ruby_headerinj_encode_value
